doc_src/language.rst

```diff
@@ -875,14 +875,15 @@ There are three kinds of variables in fish: universal, global and local variable
 - Universal variables are shared between all fish sessions a user is running on one computer.
 - Global variables are specific to the current fish session, and will never be erased unless explicitly requested by using ``set -e``.
 - Local variables are specific to the current fish session, and associated with a specific block of commands, and automatically erased when a specific block goes out of scope. A block of commands is a series of commands that begins with one of the commands ``for``, ``while`` , ``if``, ``function``, ``begin`` or ``switch``, and ends with the command ``end``.
+- Function-local variables are the same as local variables outside of one of these blocks. They go out of scope when the current function ends.
 
-Variables can be explicitly set to be universal with the ``-U`` or ``--universal`` switch, global with the ``-g`` or ``--global`` switch, or local with the ``-l`` or ``--local`` switch.  The scoping rules when creating or updating a variable are:
+Variables can be explicitly set to be universal with the ``-U`` or ``--universal`` switch, global with ``-g`` or ``--global``, local to the current block with ``-l`` or ``--local``, or local to the current function with ``-f`` or ``--function``.  The scoping rules when creating or updating a variable are:
 
 - When a scope is explicitly given, it will be used. If a variable of the same name exists in a different scope, that variable will not be changed.
 
 - When no scope is given, but a variable of that name exists, the variable of the smallest scope will be modified. The scope will not be changed.
 
-- As a special case, when no scope is given and no variable has been defined the variable will belong to the scope of the currently executing *function*. This is different from the ``--local`` flag, which would make the variable local to the current *block*.
+- As a special case, when no scope is given and no variable has been defined the variable will belong to the scope of the currently executing *function*. This is different from the ``--local`` flag, which would make the variable local to the current *block*. Outside of a function, it is global, unlike with explicit function-scope.
```

Comment:
I think this can now be simplified:

> If no scope is given and the variable is not yet defined, the variable will default to function scope if inside a function, or global scope if no function is executing.